apps/src/applab/commands.js

```diff
@@ -133,14 +133,26 @@ function apiValidateDomIdExistence(opts, funcName, varName, id, shouldExist) {
   apiValidateType(opts, funcName, varName, id, 'string');
   if (opts[validatedTypeKey] && typeof opts[validatedDomKey] === 'undefined') {
     var element = document.getElementById(id);
-    var exists = Boolean(element && divApplab.contains(element));
-    var valid = exists == shouldExist;
+
+    var existsInApplab = Boolean(element && divApplab.contains(element));
+    var existsOutsideApplab = Boolean(element && !divApplab.contains(element));
+
+    var valid = !existsOutsideApplab && (shouldExist == existsInApplab);
+
     if (!valid) {
       var line = 1 + Applab.JSInterpreter.getNearestUserCodeLine();
-      var errorString = funcName + "() " + varName +
-        " parameter refers to an id (" +id + ") which " +
-        (exists ? "already exists." : "does not exist.");
-      outputError(errorString, ErrorLevel.WARNING, line);
+      var errorString = "";
+      if (existsOutsideApplab) {
+        errorString = funcName + "() " + varName + " parameter refers to an id (" + id +
+            ") which is already in use outside of Applab. Choose a different id.";
+        outputError(errorString, ErrorLevel.ERROR, line);
+        throw new Error("Cannot create element - element ID is in use outside of applab");
+      } else {
+        errorString = funcName + "() " + varName +
+            " parameter refers to an id (" + id + ") which " +
+            (existsInApplab ? "already exists." : "does not exist.");
+        outputError(errorString, ErrorLevel.WARNING, line);
+      }
```

Comment:
Throwing an Error seems like a good approach. However, now when I try to create a runButton element, I get a redundant message:
```
ERROR: Line: 10: button() id parameter refers to an id (runButton) which is already in use outside of Applab. Choose a different id.
ERROR: Line: 10: Error: Cannot create element - element ID is in use outside of applab
```
Could you make this a single error message? I would prefer using the first one, i.e. `throw new Error(errorString)`, because...

 if I do `setText('runButton', 'text')` I get the same two error messages. The first one is accurate but the second one is misleading:
```
ERROR: Line: 1: setText() id parameter refers to an id (runButton) which is already in use outside of Applab. Choose a different id.
ERROR: Line: 1: Error: Cannot create element - element ID is in use outside of applab
```